.changelog/2145.txt

```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+data-source/mongodbatlas_advanced_cluster: Converts `replication_specs` to TypeList
+data-source/mongodbatlas_advanced_clusters: Converts `replication_specs` to TypeList
```

Comment:
Please add: "This fixes an issue in the `advanced_cluster` data source where `replication_specs` was not returned in the results".  @marcosuma -- here is the review, taken from [your discussion with customer in GH](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2076#issuecomment-2050998555).

Reply:
Thanks @JuliaMongo, used your input and changed a little bit.